src/Sulu/Bundle/AdminBundle/Resources/js/containers/FormOverlay/FormOverlay.js

```diff
@@ -0,0 +1,143 @@
+// @flow
+import React from 'react';
+import {action, computed, observable} from 'mobx';
+import type {ElementRef} from 'react';
+import {observer} from 'mobx-react';
+import Overlay from '../../components/Overlay';
+import {translate} from '../../utils';
+import Snackbar from '../../components/Snackbar';
+import Form from '../Form';
+import type {FormStoreInterface} from '../Form/types';
+import type {Size} from '../../components/Overlay/types';
+import formOverlayStyles from './formOverlay.scss';
+
+type Props = {|
+    confirmDisabled: boolean,
+    confirmLoading: boolean,
+    confirmText: string,
+    formStore: FormStoreInterface,
+    onClose: () => void,
+    onConfirm: () => void,
+    open: boolean,
+    size?: Size,
+    title: string,
+|};
+
+@observer
+class FormOverlay extends React.Component<Props> {
+    static defaultProps = {
+        confirmDisabled: false,
+        confirmLoading: false,
+    };
+
+    formRef: ?ElementRef<typeof Form>;
+
+    @observable formErrors: Array<string> = [];
+
+    @computed get confirmDisabled() {
+        const {confirmDisabled, formStore} = this.props;
+
+        return confirmDisabled || !formStore.dirty;
+    }
+
+    @computed get confirmLoading() {
+        const {confirmLoading, formStore} = this.props;
+
+        // disable confirm button while saving if formstore is instance of ResourceFormStore
+        // $FlowFixMe
+        const formStoreSaving = formStore.hasOwnProperty('saving') && formStore.saving;
```

Comment:
cannot use `formStore instanceof ResourceFormStore` here because i have not found a way to make that check work in the tests 😕